app/src/main/java/com/techcourse/service/UserService.java

```diff
@@ -23,10 +26,22 @@ public void insert(final User user) {
         userDao.insert(user);
     }
 
-    public void changePassword(final long id, final String newPassword, final String createBy) {
-        final var user = findById(id);
-        user.changePassword(newPassword);
-        userDao.update(user);
-        userHistoryDao.log(new UserHistory(user, createBy));
+    public void changePassword(final long id, final String newPassword, final String createBy) throws SQLException {
```

Comment:
여기서 SQLException을 처리하지 않고, throws한 이유가 있을까요?!
try-catch로 Exception까지 처리하고 있는데, SQLException을 throws한 이유가 궁금합니다!

Reply:
사실 어떤 방법을 써도 갈팡질팡.. 둘 다 맘에 들지 않는 갈대같은 저의 마음을 반영한 코드였습니다.. 게다가 final var connection = DataSourceConfig.getInstance().getConnection();를 try안에 넣어버리면 catch문, finally문에서 문제가 생겨서 에잇~! 하고 throw 해버린거였어요..ㅎㅎ헤헷;

이 부분 어떻게 고치는 게 좋을까요??

Reply:
해당 방식처럼 예외를 throws하면 문제 발생 시에 어디서 발생했는지 추적하기 힘들어질 것 같아요!
그래서 저는 트랜잭션에 관한 예외가 다 해당 메소드에서 처리되니까 해당 메소드 안에서 try-catch를 한번 더 써서
예외를 처리하는 건 어떨까요?!

말씀해주신 부분은 
```suggestion
DataSource dataSource = DataSourceConfig.getInstance();
        Connection connection = null;

        try {
            connection = dataSource.getConnection();
            connection.setAutoCommit(false);

           ...
```

이런 식으로 변경해보면 될 것 같습니다!